src/NetCDFIO.jl

```diff
@@ -4,140 +4,132 @@
+    # Not sure why not end+1
+    @inbounds ncdf_var[:, end] = data
 end
 
-function write_ts(self::NetCDFIO_Stats, var_name::String, data::Float64)
-    # Hack to avoid https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
-    @inbounds self.vars["timeseries"][var_name][end] = data::Float64
-    # Ideally, we remove self.vars and use:
-    # var = self.ts_grp[var_name]
-    # @inbounds var[end+1] = data :: Float64
+function write_timeseries(self::NetCDFIO_Stats, var_name::String, data::Float64)
+    # _parentname = nothing is needed to avoid https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
+    ncdf_var = NC.cfvariable(self.root_grp.group["timeseries"], var_name, _parentname = nothing)
+    # Not sure why not end+1
+    @inbounds ncdf_var[end] = data
 end
 
 function write_simulation_time(self::NetCDFIO_Stats, t::Float64)
     # # Write to profiles group
-    profile_t = self.profiles_grp["t"]
+    profile_t = self.root_grp.group["profiles"]["t"]
     @inbounds profile_t[end + 1] = t::Float64
 
     # # Write to timeseries group
-    ts_t = self.ts_grp["t"]
+    ts_t = self.root_grp.group["timeseries"]["t"]
     @inbounds ts_t[end + 1] = t::Float64
 end
 
 function KiD_output(aux, t::Float64)
 
-    # TODO: remove `vars` hack that avoids
-    # https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
-    # opening/closing files every step should be okay. #removeVarsHack
-
-    UnPack.@unpack Stats = aux
-    UnPack.@unpack ρ, p, θ_liq_ice = aux.constants
-    UnPack.@unpack T, θ_dry, q_tot, q_liq, q_ice = aux.moisture_variables
-    UnPack.@unpack q_rai, q_sno = aux.precip_variables
-
+    (; Stats) = aux
     open_files(Stats)
 
     write_simulation_time(Stats, t)
-    write_field(Stats, "density", vec(ρ), "profiles")
-    write_field(Stats, "temperature", vec(T), "profiles")
-    write_field(Stats, "pressure", vec(p), "profiles")
-
-    write_field(Stats, "theta_liq_ice", vec(θ_liq_ice), "profiles")
-    write_field(Stats, "theta_dry", vec(θ_dry), "profiles")
-
-    write_field(Stats, "q_tot", vec(q_tot), "profiles")
-    write_field(Stats, "q_liq", vec(q_liq), "profiles")
-    write_field(Stats, "q_ice", vec(q_ice), "profiles")
-    write_field(Stats, "q_rai", vec(q_rai), "profiles")
-    write_field(Stats, "q_sno", vec(q_sno), "profiles")
+
+    name_map = Dict(
+        :ρ => "density",
+        :T => "temperature",
+        :p => "pressure",
+        :θ_liq_ice => "theta_liq_ice",
+        :θ_dry => "theta_dry",
+        :q_tot => "q_tot",
+        :q_liq => "q_liq",
+        :q_ice => "q_ice",
+        :q_rai => "q_rai",
+        :q_sno => "q_sno",
+    )
+    for field in (aux.constants, aux.moisture_variables, aux.precip_variables)
+        for var in keys(name_map)
+            if var in propertynames(field)
+                prop = getproperty(field, var)
+                write_profiles(Stats, name_map[var], vec(prop))
+            end
+        end
+    end
```

Comment:
```suggestion
    NC.Dataset(Stats.nc_filename, "a") do ds
        for field in (aux.constants, aux.moisture_variables, aux.precip_variables)
            for var in keys(name_map)
                if var in propertynames(field)
                    prop = getproperty(field, var)
                    write_profiles(ds, name_map[var], vec(prop))
                end
            end
        end
    end
```
And this would be written like this

Reply:
(Note that we're not open-closing the file every time we export the data)